### ISSUE TITLE
Buffer doesn't always scroll up like it should, and sometimes it scrolls too much

### ISSUE THREAD

**Issue by kirbyfan64** [howl-editor/howl#98]
Open a file and put around 50 newlines (enough to make it auto-scroll down). If you start hitting the backspace key, the lines will be deleted, but the window won't auto-scroll back up, and the vertical scroll bar actually gets smaller.

Also, open two files and close the second view, so that there are two buffers and one view open. Now, enter enough lines in the first so that it'll auto-scroll down (I did about 30 lines), then switch to the second and do the same. When you switch back to the first, it will have scrolled too low to make room for the buffer box, and there'll be a large empty space on the bottom.

**Comment by nilnor** [comment]
@kirbyfan64 The first part of this issue should be fixed now.

I don't know exactly what you mean by the second problem, mucking around with two buffers produces nothing untoward. It might be because it was related to the first issue, but please have a look.

**Comment by kirbyfan64** [comment]
@nilnor First one works, second one still doesn't. I think the reproduction instructions were kind of suckish, so let me try again:

1. Open Howl via something like `howl x1 x2`, to open two views.
2. Close the second one `ctrl_w`. Now the first view should be open, with the buffer `x1`.
3. Enter about 40 newlines.
4. Switch buffers to `x2`.
5. Enter about 40 newlines.
6. Switch back to `x1`.

The screen should have scrolled up too high.

![howl_001](https://cloud.githubusercontent.com/assets/1690697/10051040/1793955a-61e5-11e5-88e5-56fde0853ab2.png)

**Comment by nilnor** [comment]
Ah, the missing piece was that you're switching the buffer via the buffer list. I was switching directly with `switch-to-last-hidden-buffer` so never saw this. Well, try out the latest master.
